File: sdks/node/src/resources/calendars.ts
import { ApiClient } from '../client';
import { Calendar } from '../types';

/**
 * Calendars resource for managing calendars across multiple providers
 */
export class Calendars {
  constructor(private readonly client: ApiClient) {}

  /**
   * List all calendars from all connected providers
   *
   * @returns Promise resolving to an array of calendars from all providers
   */
  async list(): Promise<Calendar[]> {
    const response = await this.client.get<Calendar[]>('/calendars');
    return response.data || [];
  }
}

File: sdks/python/mobiscroll_connect/resources/calendars.py
from __future__ import annotations

import builtins
from collections.abc import Mapping

from ..api_client import ApiClient
from ..models import Calendar


class Calendars:
    """Calendar listing across all connected providers."""

    def __init__(self, api_client: ApiClient) -> None:
        self._api = api_client

    def list(self) -> builtins.list[Calendar]:
        data = self._api.get("calendars")
        if not isinstance(data, list):
            return []
        return [Calendar.from_dict(c) for c in data if isinstance(c, Mapping)]
